trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/jdbc/mysql/MySQLDataTypeTransformer.java

```diff
@@ -11,27 +11,62 @@
 import com.datastrato.gravitino.trino.connector.GravitinoErrorCode;
 import com.datastrato.gravitino.trino.connector.util.GeneralDataTypeTransformer;
 import io.trino.spi.TrinoException;
+import io.trino.spi.type.CharType;
 
 /** Type transformer between MySQL and Trino */
 public class MySQLDataTypeTransformer extends GeneralDataTypeTransformer {
   private static final int MYSQL_CHAR_LENGTH_LIMIT = 255;
+  // 65535 / 4 = 16383.
+  private static final int MYSQL_VARCHAR_LENGTH_LIMIT = 16383;
+
+  @Override
+  public io.trino.spi.type.Type getTrinoType(Type type) {
+    if (type.name() == Name.STRING) {
+      return io.trino.spi.type.VarcharType.createUnboundedVarcharType();
+    }
+
+    return super.getTrinoType(type);
+  }
 
   @Override
   public Type getGravitinoType(io.trino.spi.type.Type type) {
-    Type gravitinoType = super.getGravitinoType(type);
-    if (gravitinoType.name() == Name.VARCHAR) {
-      if (((Types.VarCharType) gravitinoType).length() > MYSQL_CHAR_LENGTH_LIMIT) {
+    Class<? extends io.trino.spi.type.Type> typeClass = type.getClass();
+    if (typeClass == io.trino.spi.type.CharType.class) {
```

Comment:
Why don't we use the `instanceof` operator?

Reply:
I don't have any preference about `instanceof` and `getClass()`, as we will judge the type condition twice, so I think the the temporary variant and use the format `==` may be simple.